boa3/analyser/analyser.py

```diff
@@ -0,0 +1,53 @@
+import ast
+from typing import Dict
+
+from boa3.analyser.typeanalyser import TypeAnalyser
+from boa3.model.symbol import ISymbol
+from boa3.model.type.type import Type
+
+
+class Analyser(object):
+    """
+    This class is responsible for the semantic analysis of the code
+
+    :ivar symbol_table: a dictionary used to store the identifiers
+    """
+
+    def __init__(self, ast_tree: ast.AST):
+        self.symbol_table: Dict[str, ISymbol] = {}
+
+        self.ast_tree: ast.AST = ast_tree
+        self.is_analysed: bool = False
+
+        self.__include_builtins_symbols()
+
+    @staticmethod
+    def analyse(path: str):
+        """
+        Analyses the syntax of the Python code
+
+        :param path: the path of the Python file
+        :return: a boolean value that represents if the analysis was successful
+        :rtype: Analyser
+        """
+        with open(path, 'rb') as source:
+            ast_tree = ast.parse(source.read())
+
+        analyser = Analyser(ast_tree)
+        analyser.__check_types()
```

Comment:
question: the only responsability of Analyser is to call TypeAnalyser?
If true: Are we planning to reuse TypeAnalyser in a different way than Analyser do? If not, I don't see any reason to have 2 classes instead of just one.
If false: Could we add a "TODO" comment on Analyser to remember what will be done?

Reply:
Answer: No, `Analyser` is responsible to call any other analyzers that we may need.

I separated the TypeAnalyser from the Analyser because I wanted to separate the implementation of the methods inherited from the `AST` class for code readability and because this way we can include other implementations of `AST` if needed.

For converting functions it wasn't needed, but for converting variables, for example, I will have to implement a symbol table to know which and how many variables are declared in the function scope and I don't think it makes sense to include this responsibility in the `TypeAnalyser`

Reply:
Perfect